extensions/common/gcp/gcp-core/src/main/java/org/eclipse/edc/gcp/bigquery/BigQueryServiceImpl.java

```diff
@@ -0,0 +1,594 @@
+                    errorWhileAppending = true;
+                    monitor.severe("BigQuery Sink cannot parse whole input data, remaining " + buffer.getAvailableReadCount());
+                }
+
+            } catch (IOException | InterruptedException | DescriptorValidationException exception) {
+                errorWhileAppending = true;
+                monitor.severe("BigQuery Sink error while appending", exception);
+                break;
+            }
+        }
+
+        inflightRequestCount.arriveAndAwaitAdvance();
+        closeStream();
+
+        monitor.debug("BigQuery Sink requests arrived, stream " + streamWriter.getStreamName() + " for table " + target.getTableName().toString() + " closed, now finalizing..");
+        var finalizeResponse =
+                writeClient.finalizeWriteStream(streamWriter.getStreamName());
+        monitor.debug("BigQuery Sink rows written: " + finalizeResponse.getRowCount());
+
+        // Commit the data stream received only if the stream writer succeeded (transaction).
+        // TODO support option to enable commit even if the stream writer failed.
+        if (!errorWhileAppending) {
+            var commitRequest =
+                    BatchCommitWriteStreamsRequest.newBuilder()
+                    .setParent(target.getTableName().toString())
+                    .addWriteStreams(streamWriter.getStreamName())
+                    .build();
+            var commitResponse = writeClient.batchCommitWriteStreams(commitRequest);
+
+            if (!commitResponse.hasCommitTime()) {
+                var errorLogged = false;
+                for (var err : commitResponse.getStreamErrorsList()) {
+                    errorLogged = true;
+                    monitor.severe("BigQuery Sink error while committing the streams " + err.getErrorMessage());
+                }
+                if (!errorLogged) {
+                    monitor.severe("BigQuery Sink error while committing the streams");
+                }
+            } else {
+                monitor.debug("BigQuery Sink records committed successfully, write client shutting down...");
+            }
+        }
+        writeClient.shutdownNow();
+        var waitIterationCount = 0;
+        var terminated = false;
+        do {
+            try {
+                terminated = writeClient.awaitTermination(2, TimeUnit.SECONDS);
+            } catch (InterruptedException interruptedException) {
+                monitor.warning("BigQuery Sink write client shut down (interrupted)");
+            }
+            waitIterationCount++;
+        } while (!terminated && waitIterationCount < 3);
+
+        if (terminated) {
+            monitor.info("BigQuery Sink write client shut down");
+        } else {
+            monitor.warning("BigQuery Sink write client NOT shut down after timeout");
+        }
+    }
+
+    @Override
+    public boolean tableExists(BigQueryTarget target) {
```

Comment:
target is already a field in the class, keep either this or that

Reply:
Removed method parameter and keeping the field